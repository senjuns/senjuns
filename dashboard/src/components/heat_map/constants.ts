// TODO - Update the resource path to real room image.
export const ROOM_IMAGE_PATH = '/zone_map.png';

// Colors that are being used to display the heatmap in gradient.
export const COLOR_SCALE_VALUES: Array<[number, string]> = [
  [0, '#0046B3'], // Blue
  [0.33, '#6FC153'], // Green
  [0.67, '#FFB94C'], // Yellow
  [1, '#E83339'], // Red
];

export const FRAME_TRANSITION_DURATION = 500;
