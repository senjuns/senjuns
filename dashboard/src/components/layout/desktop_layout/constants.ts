// Header height of the desktop layout
export const HEADER_HEIGHT = 110;

// Sidebar width of the desktop layout
export const SIDEBAR_WIDTH = 110;
