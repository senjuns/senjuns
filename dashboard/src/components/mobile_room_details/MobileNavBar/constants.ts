export const RoomDetailTab = {
  Overview: 0,
  HeatMap: 1,
  LineChart: 2,
  PhotoFeed: 3,
};
