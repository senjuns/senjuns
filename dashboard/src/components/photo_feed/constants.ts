/**
 * TODO(NL-1391): improve the logic to remove the hardcode
 * These constants regarding platform specific need to be updated/removed sometime later in NL-1391.
 *
 * When the app is rendered on the mobile users, there are some headers and footers which need to be displayed.
 * And 280 seems to be the correct size for the mobile platform in order to display the photo feed in the whole device.
 */
export const MOBILE_PLATFORM_EXTRA_HEIGHT = 180;
/**
 * The height and width are just placeholders for non-mobile app users and we need to match this with the figma.
 */
export const NON_MOBILE_PLATFORM_PRESENTATION_HEIGHT = 400;
export const NON_MOBILE_PLATFORM_PRESENTATION_WIDTH = 400;

export const ZONE_MAP_URL = '/zone_map.png';
/**
 * The mini map height to be rendered on the screen.
 */
export const MINI_MAP_HEIGHT = 40;

/**
 * The zoom level limitations of the photo feed.
 */
export const ZOOM_LEVELS = {
  MAX: 2,
  NORMAL: 1,
  MIN: 0.5,
};

/**
 * When the section is unvisited, there needs to be opacity applied on it.
 */
export const UNVISITED_SECTION_OPACITY = 0.3;
