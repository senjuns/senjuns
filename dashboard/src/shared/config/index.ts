import Cookies from 'js-cookie';
// import { FEATURE_ENABLED } from '../../shared/constants';
import { FEATURE_FLAGS } from '../../shared/interfaces';
// @ts-ignore Needs to be ignored as it is dynamically created.
import config from './config';

const defaultConfig: any = config;

if (Cookies.get(FEATURE_FLAGS.DEBUG) === 'FEATURE_ENABLED') {
  console.log(`process.env: ${JSON.stringify(process.env, null, 2)}`);
  console.log(`defaultConfig: ${JSON.stringify(defaultConfig, null, 2)}`);
}

export default defaultConfig;
