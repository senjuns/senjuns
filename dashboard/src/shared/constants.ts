import {
  MeasurementTypes,
  MeasurementStatisticsTypes,
  TPasswordRule,
} from '../shared/interfaces';

export const MeasurementUnit = {
  CelciusDegree: '°C',
  FahrenheitDegree: '°F',
  Percent: '%',
  KP: 'kP',
  mph: 'mph',
  ppm: 'ppm',
  DLI: 'DLI',
  μMol_m2_s: 'μMol/m2/s',
};

export const NotificationLevel = {
  Nominal: 'nominal',
  Warning: 'warning',
  Error: 'error',
};

export const ScreenSize = {
  xs: 576,
  sm: 768,
  md: 992,
  lg: 1200,
  xl: 1400,
};

export const Colors = {
  brightBlue100: '#60C3E3',
  brightBlue80: '#7FCEE8',
  brightBlue60: '#9FDAEE',
  grey100: '#E5E5E5',
  grey80: '#EAEAEA',
  grey60: '#EFEFEF',
  black: '#000000',
  dark1: '#383838',
  dark7: '#808080',
  dark8: '#B4B4B4',
  dark9: '#CFCFCF',
  green1: '#56C254',
  grey5: '#E0E0E0',
  hb1: '#82879A',
  hb2: '#9BA0B6',
  hb3: '#C3C7D9',
  hb4: '#ECEDF1',
  hb5: '#F0F1F4',
  hb6: '#F8F8FC',
  orange1: '#DD382F',
  orange2: '#E86339',
  orange3: '#F48E42',
  orange4: '#FFB94C',
  white: '#ffffff',
  linearGradientedOrange: 'linear-gradient(90deg, #E86339 0%, #FFB94C 100%)',
};

export const Sizes = {
  xxSmall: 4,
  xSmall: 8,
  small: 12,
  medium: 16,
  large: 20,
  xLarge: 24,
  xxLarge: 32,
};

export const Fonts = {
  h1: { size: 48, style: 'normal', weight: 600 },
  h2: { size: 40, style: 'normal', weight: 600 },
  h3: { size: 36, style: 'normal', weight: 600 },
  h4: { size: 24, style: 'normal', weight: 600 },
  h5: { size: 20, style: 'normal', weight: 600 },
  h6: { size: 18, style: 'normal', weight: 600 },
  body1: { size: 16, style: 'normal', weight: 'normal' },
  body2: { size: 14, style: 'normal', weight: 'normal' },
};

export const MEASUREMENT_OPTION_DETAILS = {
  [MeasurementTypes.Light]: {
    statisticsKey: MeasurementStatisticsTypes.AirTemperature,
    unit: MeasurementUnit.FahrenheitDegree,
  },
  [MeasurementTypes.Temperature]: {
    statisticsKey: MeasurementStatisticsTypes.AirTemperature,
    unit: MeasurementUnit.FahrenheitDegree,
  },
  [MeasurementTypes.Humidity]: {
    statisticsKey: MeasurementStatisticsTypes.RelativeHumidity,
    unit: MeasurementUnit.Percent,
  },
  [MeasurementTypes.LeafVpd]: {
    statisticsKey: MeasurementStatisticsTypes.LeafVpd,
    unit: MeasurementUnit.KP,
  },
  [MeasurementTypes.Vpd]: {
    statisticsKey: MeasurementStatisticsTypes.AirVpd,
    unit: MeasurementUnit.KP,
  },
  [MeasurementTypes.Airflow]: {
    statisticsKey: MeasurementStatisticsTypes.AirFlow,
    unit: MeasurementUnit.mph,
  },
  [MeasurementTypes.Co2]: {
    statisticsKey: MeasurementStatisticsTypes.AirCo2,
    unit: MeasurementUnit.ppm,
  },
  [MeasurementTypes.Ppfd]: {
    statisticsKey: MeasurementStatisticsTypes.Par,
    unit: MeasurementUnit.μMol_m2_s,
  },
  [MeasurementTypes.LeafTemperature]: {
    statisticsKey: MeasurementStatisticsTypes.LeafTemperature,
    unit: MeasurementUnit.FahrenheitDegree,
  },
};

export const APP_URL = {
  login: '/login',
  forgot: '/forgot',
  reset: '/reset',
  notFound: '/not-found',
  resetPassword: (username: string) => `/reset?username=${username}`,
  userHome: '/user/home',
  userAbout: '/user/about',
  home: '/',
  roomDetails: (id: number) => `/room-details/${id}`,
  roomDetailsOverview: (id: number) => `/room-details/${id}/overview`,
  roomDetailsHeatMap: (id: number) => `/room-details/${id}/heat-map`,
  roomDetailsLineChart: (id: number) => `/room-details/${id}/line-chart`,
  roomDetailsPhotoFeed: (id: number) => `/room-details/${id}/photo-feed`,
};

export enum ROOM_DETAIL_TABS {
  OVERVIEW = 'overview-tab',
  LINE_CHART = 'line-chart-tab',
  HEAT_MAP = 'heat-map-tab',
  PHOTO_FEED = 'photo-feed-tab',
}

export const AWS_API_CODE = {
  // will return when username or password incorrect or when user does not exists.
  user_not_found_exception: 'UserNotFoundException',
};

export const PASSWORD_RULES: TPasswordRule[] = [
  {
    label: 'Password must match',
    check: (password: string, confirmPassword: string): boolean =>
      password === confirmPassword,
  },
  {
    label: 'At least 8 characters',
    check: (password: string): boolean => password.length >= 8,
  },
  {
    label: 'Must contain at least one upper case letter',
    check: (password: string): boolean => /[A-Z]/.test(password),
  },
  {
    label: 'Must contain at least one special character',
    check: (password: string): boolean => /[#?!@$%^&*-]/.test(password),
  },
  {
    label: 'Must contain at least one number',
    check: (password: string): boolean => /[0-9]/.test(password),
  },
];

export const FEATURE_ENABLED = 'FEATURE_ENABLED';

export const NULL_VALUE = -1;

export const Z_INDEXES = {
  DROPDOWN: 10001,
};
