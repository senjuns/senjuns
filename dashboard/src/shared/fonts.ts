import { css } from 'styled-components';

export const InterBoldMirage16px = css`
  color: var(--mirage);
  font-family: var(--font-family-inter);
  font-size: var(--font-size-s);
  font-weight: 700;
  font-style: normal;
`;

export const Poppins18 = css`
  font-family: var(--font-family-poppins);
  font-size: var(--font-size-l2);
  letter-spacing: 0;
  font-weight: 400;
  font-style: normal;
`;

export const Poppins22 = css`
  font-family: var(--font-family-roboto);
  font-size: var(--font-size-22px);
  letter-spacing: 0;
  font-weight: 400;
  font-style: normal;
`;

export const Poppins26 = css`
  font-family: var(--font-family-poppins);
  font-size: var(--font-size-xxl2);
  letter-spacing: 0;
  font-weight: 400;
  font-style: normal;
`;

export const RobotoBoldBlack22px = css`
  color: var(--black);
  font-family: var(--font-family-roboto);
  font-size: var(--font-size-22px);
  font-weight: 700;
  font-style: normal;
`;
