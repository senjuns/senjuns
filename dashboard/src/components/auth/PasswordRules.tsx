import { FC } from 'react';
import styled from 'styled-components';

import { ReactComponent as Check } from '../../assets/svg/check.svg';
import { Typography } from '../../components/common';
import { Colors, Sizes } from '../../shared/constants';
import { TPasswordRule } from '../../shared/interfaces';

interface PasswordRulesProps {
  /**
   * ConfirmPassword
   */
  confirmPassword: string;
  /**
   * Password
   */
  password: string;
  /**
   * Rules to check in the component.
   */
  rules: TPasswordRule[];
}

/**
 * PasswordRules component.
 *
 * @param {PasswordRulesProps} - The component props.
 * @returns {JSX.Element} - The rendered component.
 */
export const PasswordRules: FC<PasswordRulesProps> = ({
  confirmPassword,
  password,
  rules,
}) => {
  return (
    <RulesContainer>
      {rules.map((rule) => (
        <RuleItem key={rule.label}>
          <CheckIcon disabled={!rule.check(password, confirmPassword)}>
            <Check />
          </CheckIcon>
          <Typography variant="body1">{rule.label}</Typography>
        </RuleItem>
      ))}
    </RulesContainer>
  );
};

const RulesContainer = styled.ul`
  display: flex;
  flex-direction: column;
  gap: 14px;
  list-style: none;
  margin: 0;
  padding: 0;
  margin-top: 30px;
`;

const RuleItem = styled.li`
  display: flex;
  align-items: center;
  flex-direction: row;

  &:not(:last-type-of) {
    margin-bottom: ${Sizes.small}px;
  }
`;

interface CheckIconProps {
  disabled: boolean;
}

const CheckIcon = styled.span<CheckIconProps>`
  color: ${(props) => (props.disabled ? Colors.hb4 : Colors.green1)};
  margin-right: ${Sizes.medium}px;
`;
