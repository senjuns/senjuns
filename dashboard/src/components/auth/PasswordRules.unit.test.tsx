// eslint-disable-next-line import/no-extraneous-dependencies
import * as ShallowRenderer from 'react-test-renderer/shallow';

import { PASSWORD_RULES } from '../../shared/constants';

import { PasswordRules } from './PasswordRules';

describe('<PasswordRules />', () => {
  it('renders correctly.', async () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <PasswordRules
        confirmPassword="confirm-password"
        password="my-password"
        rules={PASSWORD_RULES}
      >
        Auth Layout
      </PasswordRules>
    );

    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
