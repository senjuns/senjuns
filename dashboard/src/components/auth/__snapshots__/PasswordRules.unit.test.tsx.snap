// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordRules /> renders correctly. 1`] = `
<styled.ul>
  <styled.li>
    <styled.span
      disabled={true}
    >
      <ForwardRef(SvgCheck) />
    </styled.span>
    <Typography
      variant="body1"
    >
      Password must match
    </Typography>
  </styled.li>
  <styled.li>
    <styled.span
      disabled={false}
    >
      <ForwardRef(SvgCheck) />
    </styled.span>
    <Typography
      variant="body1"
    >
      At least 8 characters
    </Typography>
  </styled.li>
  <styled.li>
    <styled.span
      disabled={true}
    >
      <ForwardRef(SvgCheck) />
    </styled.span>
    <Typography
      variant="body1"
    >
      Must contain at least one upper case letter
    </Typography>
  </styled.li>
  <styled.li>
    <styled.span
      disabled={false}
    >
      <ForwardRef(SvgCheck) />
    </styled.span>
    <Typography
      variant="body1"
    >
      Must contain at least one special character
    </Typography>
  </styled.li>
  <styled.li>
    <styled.span
      disabled={true}
    >
      <ForwardRef(SvgCheck) />
    </styled.span>
    <Typography
      variant="body1"
    >
      Must contain at least one number
    </Typography>
  </styled.li>
</styled.ul>
`;
